runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/Boss.java

```diff
@@ -38,20 +38,33 @@
 	CALLISTO(NpcID.CALLISTO, NpcID.CALLISTO_6609),
 	CERBERUS(NpcID.CERBERUS, NpcID.CERBERUS_5863, NpcID.CERBERUS_5866),
 	CHAOS_ELEMENTAL(NpcID.CHAOS_ELEMENTAL, NpcID.CHAOS_ELEMENTAL_6505),
-	CORPOREAL_BEAST(NpcID.CORPOREAL_BEAST),
+	CORPOREAL_BEAST(new Integer[]{NpcID.DARK_ENERGY_CORE}, NpcID.CORPOREAL_BEAST),
 	GENERAL_GRAARDOR(NpcID.GENERAL_GRAARDOR, NpcID.GENERAL_GRAARDOR_6494),
 	GIANT_MOLE(NpcID.GIANT_MOLE, NpcID.GIANT_MOLE_6499),
 	KALPHITE_QUEEN(NpcID.KALPHITE_QUEEN, NpcID.KALPHITE_QUEEN_963, NpcID.KALPHITE_QUEEN_965, NpcID.KALPHITE_QUEEN_4303, NpcID.KALPHITE_QUEEN_4304, NpcID.KALPHITE_QUEEN_6500, NpcID.KALPHITE_QUEEN_6501),
 	KING_BLACK_DRAGON(NpcID.KING_BLACK_DRAGON, NpcID.KING_BLACK_DRAGON_2642, NpcID.KING_BLACK_DRAGON_6502),
 	KRIL_TSUROTH(NpcID.KRIL_TSUTSAROTH, NpcID.KRIL_TSUTSAROTH_6495),
+	THE_MAIDEN_OF_SUGADINTI(new Integer[]{NpcID.BLOOD_SPAWN, NpcID.NYLOCAS_MATOMENOS}, NpcID.THE_MAIDEN_OF_SUGADINTI, NpcID.THE_MAIDEN_OF_SUGADINTI_8361, NpcID.THE_MAIDEN_OF_SUGADINTI_8362, NpcID.THE_MAIDEN_OF_SUGADINTI_8363, NpcID.THE_MAIDEN_OF_SUGADINTI_8364, NpcID.THE_MAIDEN_OF_SUGADINTI_8365),
+	NYLOCAS_VASILIAS(NpcID.NYLOCAS_VASILIAS, NpcID.NYLOCAS_VASILIAS_8355, NpcID.NYLOCAS_VASILIAS_8356, NpcID.NYLOCAS_VASILIAS_8357),
 	VENETENATIS(NpcID.VENENATIS, NpcID.VENENATIS_6610),
-	VETION(NpcID.VETION, NpcID.VETION_REBORN);
+	VETION(NpcID.VETION, NpcID.VETION_REBORN),
+	VORKATH(new Integer[]{NpcID.ZOMBIFIED_SPAWN, NpcID.ZOMBIFIED_SPAWN_8063}, NpcID.VORKATH, NpcID.VORKATH_8058, NpcID.VORKATH_8059, NpcID.VORKATH_8060, NpcID.VORKATH_8061);
 
 	private final Set<Integer> ids;
+	private final Set<Integer> minionIds;
 
 	Boss(Integer... ids)
 	{
 		this.ids = Sets.newHashSet(ids);
+		this.minionIds = Sets.newHashSet(new Integer[]{});
+	}
+
+	Boss(Integer[] minionIds, Integer... bossIds)
+	{
+		Set<Integer> npcIds = Sets.newHashSet(minionIds);
```

Comment:
Using this as the basis for the boss ID set means that hitting a minion with a special attack would generate a special attack infobox, which is not desirable.